src/containers/tokens/index.js

```diff
@@ -123,6 +164,28 @@ class Tokens extends PureComponent {
     buyPNK(decimalStringToWeiBN(amount).toString())
   }
 
+  handleBuyPNKFromBondingCurveForm = formData => {
+    const { buyPNKFromBondingCurve } = this.props
+    const { amountOfETH } = formData
+    buyPNKFromBondingCurve(decimalStringToWeiBN(amountOfETH).toString())
```

Comment:
You can do this in the saga and avoid a new function.

Reply:
Following patterns (`handleXxx` functions before it).

Reply:
You can just pass `buyPNKFromBondingCurve` as the handler.